packages/@aws-cdk/aws-apigatewayv2/README.md

```diff
@@ -134,3 +134,55 @@ If you omit the `stageName` will create a `$default` stage. A `$default` stage i
 the API's URL - `https://{api_id}.execute-api.{region}.amazonaws.com/`.
 
 Note that, `HttpApi` will always creates a `$default` stage, unless the `createDefaultStage` property is unset.
+
+
+
+### Custom Domain Name
+
+Custom domain names are simpler and more intuitive URLs that you can provide to your API users. Each custom domain name can be associated with a single API stage. Use `HttpApiMapping` to map the your custom name to a specific stage. 
+
+The code snippet below configures a custom domain for your API and creates a default `HttpApiMapping` that maps the custom domain to the `$default` stage of the API.
+
+```ts
+const api = new HttpApi(stack, 'HttpProxyApi', {
+  defaultIntegration: new HttpProxyIntegration({ url }),
+  domainName: {
+    certificate: acmCertificateForExampleCom,
+    domainName: 'example.com',
+  }
+});
+```
```

Comment:
It seems that `DomainName` is not tied to any specific `Api` and can be reused across multiple Stages in multiple Apis. `ApiMapping` is the system to associate a `DomainName` with a specific `Api`.

Given this, would it be better to model something like this instead -

```ts
const domainName = new DomainName(...);

const api = new HttpApi(stack, 'HttpApi', {
  ...
  defaultDomainMapping: {
    domain: domainName,
    mappingKey: 'prod',
  },
});

api.addStage('beta', {
  stageName: 'beta',
  domainMapping: {
    domain: domainName,
    mappingKey: 'beta',
  },
});
```

This splits out `DomainName` as a construct that can stand without an `IApi` instance and can be used even across HTTP and Websocket APIs, if defined within the same stack.